components/org.wso2.carbon.metrics.impl/src/main/java/org/wso2/carbon/metrics/impl/MetricServiceImpl.java

```diff
@@ -662,6 +662,13 @@ public Meter getMeter(String name) throws MetricNotFoundException {
         }
     }
 
+    @Override
+    public boolean removeMetric(String name) {
+        metricsMap.remove(name);
+        metricsCollections.remove(name);
+        return metricRegistry.remove(name);
```

Comment:
This should be metricRegistry.remove(name) && (metricsMap.remove(name) || metricsCollections.remove(name))